server/middlewares/authentication.js

```diff
@@ -0,0 +1,54 @@
+import jwt from 'jsonwebtoken';
+import db from '../models';
+
+/**
+ * Secret token for jsonwebtoken
+ */
+const secret = process.env.SECRET || 'my secret key';
+
+const Authentication = {
+
+  /**
+   * verifyToken - Verifies if a token supplied is valid or not
+   *
+   * @param  {Object} req  Request Object
+   * @param  {Object} res  Response Object
+   * @param  {Object} next
+   * @returns {Object} Response Status
+   */
+  verifyToken(req, res, next) {
+    const token = req.headers.authorization || req.headers['x-access-token'];
+    if (!token) {
+      return res.status(401).send({ message: 'Authentication required to access this route!' });
+    }
+
+    jwt.verify(token, secret, (err, decoded) => {
+      if (err) {
+        return res.status(401).send({ message: 'Authentication failed due to invalid token!' });
+      }
+      req.decoded = decoded;
+      next();
+    });
+  },
+
+  /**
+   * verifyAdmin - Verifies that the user role supplied is an admin
+   *
+   * @param  {Object} req  Request Object
+   * @param  {Object} res  Response Object
+   * @param  {Object} next
+   * @returns {Object} Response Object
+   */
+  verifyAdmin(req, res, next) {
+    db.Role.findById(req.decoded.roleId)
+      .then((role) => {
+        if (role.title === 'admin') {
+          next();
+        } else {
+          return res.status(403).send({ message: 'Access Restricted; You are not an admin!' });
```

Comment:
Ditto more generic message `You are not authorized to view this page`.

All these generic message things is just to make your API look more professional, you really don't have to change them.